### PR TITLE
DP-30820: Stop collection_all View from spamming the watchdog log

### PR DESCRIPTION
**Pull request by arthurbaghdas**
**Description:**
Explain the technical implementation of the work done.

**Jira:** (Skip unless you are MA staff)
DP-30820

**To Test:**
- [ ] Add steps to test this feature

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

**Comment by joeg8612** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Add a changelog YAML file to this PR</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b65e344dc21fb9b7192df4d675b45ebf6688781e
</p>

File: docroot/modules/custom/mass_views/mass_views.module [hunk 1]
```diff
@@ -229,3 +229,27 @@ function mass_views_do_invalidate($entity): void {
     Cache::invalidateTags($tags);
   }
 }
+
+/**
+ * Implements hook_views_pre_render().
+ */
```
Comment (weitzman) [inline docroot/modules/custom/mass_views/mass_views.module:237]: Unclear to me what this function is doing. Please add code comment.

File: docroot/modules/custom/mass_views/mass_views.module [hunk 2]
```diff
@@ -229,3 +229,27 @@ function mass_views_do_invalidate($entity): void {
...
+function mass_views_views_pre_render(ViewExecutable $view) {
+  if ($view->id() == 'collection_all') {
+    foreach ($view->result as $row) {
+      // Check if the row entity is a media entity
+      if ($row->_entity == NULL && $row->entity_type == 'media') {
+        // Access the fields of the row
```
Comment (weitzman) [inline docroot/modules/custom/mass_views/mass_views.module]: What do we mean by 'access'? Did you mean unset? Would be good to say why we are doing it, and not just what we are doing.

File: docroot/modules/custom/mass_content/mass_content.module [hunk 3]
```diff
@@ -819,10 +819,11 @@ function mass_content_query_collection_all_check_sort_value($value = '') {
  */
 function _mass_content_collection_all_only_future_events(QueryPluginBase $query) {
   $collection = mass_content_get_collection_from_current_page();
-  $only_future = $collection->field_show_only_future_events->value;
-
-  if (!$only_future) {
-    return;
+  if ($collection) {
+    $only_future = $collection->field_show_only_future_events->value;
```
Comment (weitzman) [inline docroot/modules/custom/mass_content/mass_content.module]: In the future, please create a method in a bundle class when you need to access a field like field_show_only_future_events. This makes the code more robust. I know this code pre-exists this PR but its important for us to modernize the codebase as we implement features and bug fixes. An example of a bundle class using PR is https://github.com/massgov/openmass/pull/2336

File: docroot/modules/custom/mass_views/mass_views.module (hunk 2 above)
Comment (arthurbaghdas) [reply docroot/modules/custom/mass_views/mass_views.module]: Added a more detailed explanation.

File: docroot/modules/custom/mass_views/mass_views.module (hunk 1 above)
Comment (arthurbaghdas) [reply docroot/modules/custom/mass_views/mass_views.module:237]: Added some more info to describe the usage.

File: docroot/modules/custom/mass_content/mass_content.module (hunk 3 above)
Comment (arthurbaghdas) [reply docroot/modules/custom/mass_content/mass_content.module]: Refactored the code as suggested, thanks.

**Comment by weitzman** [comment]
Approved. I'm fixing PHPCS buglet and will push

**Pull request merged**